pkg/controller/config/extensions/imagereplacement/helpers_test.go

```diff
@@ -0,0 +1,155 @@
+	},
+
+	{
+		name: "ImageinArgsandContainerMultiple",
+		containers: []corev1.Container{{
+			Name:  "tekton-pipelines-controller",
+			Image: "quay.io/openshift-pipeline/tektoncd-pipeline-webhook:v0.4.0",
+			Args: []string{"-kubeconfig-writer-image", "quay.io/openshift-pipeline/tektoncd-pipeline-kubeconfigwriter:v0.4.0",
+				"-creds-image", "quay.io/openshift-pipeline/tektoncd-pipeline-creds-init:v0.4.0"},
+		}},
+		registry: tektonv1alpha1.Registry{
+			Override: map[string]string{
+				"tekton-pipelines-controller": "quay.io/openshift-pipeline/tektoncd-pipeline-webhook:new-tag",
+				"-kubeconfig-writer-image":    "quay.io/openshift-pipeline/tektoncd-pipeline-kubeconfigwriter:new-tag",
+				"-creds-image":                "quay.io/openshift-pipeline/tektoncd-pipeline-creds-init:new-tag",
+			},
+		},
+		argsIndex: []int8{1, 3},
+		expected: []string{"quay.io/openshift-pipeline/tektoncd-pipeline-webhook:new-tag",
+			"quay.io/openshift-pipeline/tektoncd-pipeline-kubeconfigwriter:new-tag",
+			"quay.io/openshift-pipeline/tektoncd-pipeline-creds-init:new-tag"},
+	},
+	{
+		name: "replaceNothing",
+		containers: []corev1.Container{{
+			Name:  "tekton-pipelines-controller",
+			Image: "quay.io/openshift-pipeline/tektoncd-pipeline-webhook:v0.4.0",
+			Args: []string{"-kubeconfig-writer-image", "quay.io/openshift-pipeline/tektoncd-pipeline-kubeconfigwriter:v0.4.0",
+				"-creds-image", "quay.io/openshift-pipeline/tektoncd-pipeline-creds-init:v0.4.0"},
+		}},
+		registry: tektonv1alpha1.Registry{
+			Override: map[string]string{},
+		},
+		argsIndex: []int8{1, 3},
+		expected: []string{"quay.io/openshift-pipeline/tektoncd-pipeline-webhook:v0.4.0",
+			"quay.io/openshift-pipeline/tektoncd-pipeline-kubeconfigwriter:v0.4.0",
+			"quay.io/openshift-pipeline/tektoncd-pipeline-creds-init:v0.4.0"},
+	},
+}
+
+func TestUpdateDeploymentImage(t *testing.T) {
+	for _, tt := range updateDeploymentImageTests {
+		t.Run(tt.name, func(t *testing.T) {
+			runUpdateDeploymentImageTest(t, tt)
+		})
+	}
+}
+
+func runUpdateDeploymentImageTest(t *testing.T, tt imageReplacementTest) {
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: tt.name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: tt.containers,
+				},
+			},
+		},
+	}
+	log := logf.Log.WithName(tt.name)
+	logf.SetLogger(logf.ZapLogger(true))
+
+	if err := UpdateDeployment(&deployment, &tt.registry, log); err != nil {
+		t.Fatalf("Expected success but got error: %s", err)
+	}
+
+	expecteds := tt.expected
+	// Assert container[*].image
+	assertEqual(t, deployment.Spec.Template.Spec.Containers[0].Image, expecteds[0])
+	expecteds = expecteds[1:]
+	// Assert container[*].args
+	for i, argsIndex := range tt.argsIndex {
+		assertEqual(t, deployment.Spec.Template.Spec.Containers[0].Args[argsIndex], expecteds[i])
+	}
+
+}
+
+func assertEqual(t *testing.T, actual, expected string) {
+	if actual == expected {
```

Comment:
wouldn't it be better to add
`t.Helper()` as the first line in the function `assertEqual`

Reply:
ok